src/index/base.cpp

```diff
@@ -75,11 +75,7 @@ bool BaseIndex::Init()
         if (!m_best_block_index) {
             // index is not built yet
             // make sure we have all block data back to the genesis
-            const CBlockIndex* block = active_chain.Tip();
-            while (block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA)) {
-                block = block->pprev;
-            }
-            prune_violation = block != active_chain.Genesis();
+            prune_violation = node::GetFirstStoredBlock(active_chain.Tip()) != active_chain.Genesis();
```

Comment:
Just noticing that the reason this cannot not trigger the added assert in `GetFirstStoredBlock()` is that if `active_chain.Tip()` is nullptr (which happens on first startup), then `m_synced` will be determined to be true, and we don't get to this point (#22485 documents this). I think that this is a bit fragile, but doesn't need to be addressed in this PR.